plutus-core/testlib/PlutusCore/Generators/QuickCheck/Builtin.hs

```diff
@@ -245,7 +248,71 @@ instance Arbitrary Data where
     arbitrary = arbitraryBuiltin
     shrink = shrinkBuiltin
 
-instance ArbitraryBuiltin Value
+---------------Generator for Builtin Value---------------
+
+{-| Return how many candidates to randomly choose from to fill the given number of cells. For
+example, if we only need to fill a single cell, we choose from 6 different candidates, and if we
+need to fill 5 cells, we choose from 11 candidates.
+
+>>> map (\i -> (i, toCellCandidatesNumber i)) [1..13]
+[(1,6),(2,6),(3,6),(4,8),(5,11),(6,14),(7,18),(8,22),(9,27),(10,31),(11,36),(12,41),(13,46)]
+-}
+toCellCandidatesNumber :: Int -> Int
+toCellCandidatesNumber i = max 6 . floor @Double $ fromIntegral i ** 1.5
+
+{-| Generate a 'ByteString' by picking one of the predetermined ones, given a number of
+cells to fill (see 'toCellCandidatesNumber'). The idea is that we want to occasionally generate
+the same 'CurrencySymbol' or 'TokenName' for different 'Value's to have decent test coverage,
+hence to make name clashing more likely we pick from a predetermined set of
+'ByteString's. Otherwise the chance of generating the same 'ByteString' for two
+different 'Value's would be virtually zero.
+-}
+genShortHex :: Int -> Gen ByteString
+genShortHex i =
+  Base16.encode . BC.pack . show <$> elements [0 .. toCellCandidatesNumber i]
```

Comment:
This was naive on my part BTW, as you noted we should generate large bytestrings too and as Yura noted we should generate regular token and currency names. So this generator that I created is no good at all.